### PR TITLE
fix: parse tsx files with the tseslint parser

### PR DESCRIPTION
**Pull request by vhscom**
Adds support for TSX parsing and documents type used given ESLint now has two config types. Information about the new FlatConfig from Nicholas Zakas can be found here: https://eslint.org/blog/2022/08/new-config-system-part-2/

Ref: https://typescript-eslint.io/packages/typescript-eslint/ on the set-up.

### CODE REVIEW

File: web-ui/eslint.config.js [hunk 1]
```diff
@@ -31,6 +41,13 @@ export default [
       'react-hooks/exhaustive-deps': 'warn',
       ...vitestPlugin.configs.recommended.rules,
       'vitest/expect-expect': 'off',
+      '@typescript-eslint/no-unused-vars': 'off',
```
Comment (mvolkmann) [inline web-ui/eslint.config.js:44]: Why do we want to turn these rules off?

File: web-ui/eslint.config.js [hunk 2]
```diff
@@ -31,6 +41,13 @@ export default [
...
+      '@typescript-eslint/ban-ts-comment': 'off'
     }
+  },
+  {
+    // disable type-aware linting on JS and JSX files
```
Comment (mvolkmann) [inline web-ui/eslint.config.js:49]: Why do we want to do this?

File: web-ui/yarn.lock [hunk 3]
```diff
@@ -1363,7 +1363,7 @@
   dependencies:
     eslint-visitor-keys "^3.3.0"
 
-"@eslint-community/regexpp@^4.6.1":
+"@eslint-community/regexpp@^4.10.0", "@eslint-community/regexpp@^4.6.1":
```
Comment (mvolkmann) [inline web-ui/yarn.lock:1366]: Are we using two versions of this same package?

File: web-ui/eslint.config.js (hunk 1 above)
Comment (vhscom) [reply web-ui/eslint.config.js]: That was protoype code. It's been removed.

File: web-ui/eslint.config.js (hunk 2 above)
Comment (vhscom) [reply web-ui/eslint.config.js:43]: That was protoype code. It's been removed.

File: web-ui/yarn.lock (hunk 3 above)
Comment (vhscom) [reply web-ui/yarn.lock:1366]: We're not, but one of our transient dependencies may be.

File: docs/getting-started/running/index.md [hunk 4]
```diff
@@ -63,10 +63,14 @@ To run the UI tests, use the following command:
 yarn --cwd web-ui test
 ```
 
-Or simply `cd` to the `web-ui` directory and run `yarn test`.
+Or simply `cd` to the `web-ui` directory and run `yarn test`. Tests using Snapshots are likely to fail with component updates. Those which fail to meet their Snapshot will be marked as failed. You can update the Snapshot by running the following command:
+
+```shell
+yarn --cwd web-ui test -u
+```
```
Comment (mvolkmann) [inline docs/getting-started/running/index.md:70]: I'm thinking you should also mention the option of just pressing the "u" key after failed snapshot tests are reported when the tests are run with "yarn test" (runs in watch mode).

**Review by ZacharyKlein — APPROVED** [review]

**Review by mvolkmann — APPROVED** [review]

**Review by jackkeller — APPROVED** [review]
LGTM 🚀

**Review by ZacharyKlein — APPROVED** [review]

**Review by mvolkmann — APPROVED** [review]

**Pull request merged**